scrunch/datasets.py

```diff
@@ -1593,7 +1593,47 @@ def create_categorical(self, categories, alias, name, multiple, description=''):
                 categories, alias=alias, name=name, description=description)
 
 
-class Variable(ReadOnly):
+class DatasetSubvariablesMixin(collections.Mapping):
+    """
+    Handles a variable subvariables iteration in a dict-like way
+    """
+
+    def __getitem__(self, item):
+        # Check if the attribute corresponds to a variable alias
+        subvariable = self.resource.subvariables.by('alias').get(item)
+        if subvariable is None:
+            # subvariable doesn't exists, must raise a ValueError
+            raise ValueError('Variable %s has no subvariable %s' % (
+                self.resource['name'], item))
+        # subvariable exists!, return the subvariable Instance
+        return Variable(subvariable, self)
+
+    def __iter__(self):
+        for var in self.resource.subvariables.index.items():
+            yield var
+
+    def __len__(self):
+        return len(self.resource.subvariables.index.items())
```

Comment:
No need to do `.items()` here, you're generating a new list with the items when you only want the length, doing len on the index will do the len on the dict which will be the same result without creating the intermediate items list.